### DISCUSSION TITLE
Switching Phones or adding followers - Transfer glucose level alerts

### DISCUSSION THREAD

**Discussion by roco5** [NightscoutFoundation/xDrip#2368] [General]
Is there a way to transfer glucose level alerts, when switching phones, adding followers, or rebuilding an existing installation.  I was hopeful the glucose level alerts settings would come over with the All Settings QR code, but they did not.  I have not had any luck exporting and reinstalling the DB as the export never gets written to my phone's main memory and therefore cannot be shared or even sent to google drive.  

Any assistance here would be appreciated.  Thank you for such an amazing and life changing software.

**Comment by roco5** [comment]
![image](https://user-images.githubusercontent.com/6946920/193862656-ad4e9762-dd26-42cc-a170-3413ece2a079.png)

**Comment by Navid200** [comment]
First, let's talk about the issue you are describing with export.
Please provide the phone brand model that you are experiencing this problem on.
Also, please provide a screenshot of the classic status page so that we can see which version you are using:
https://navid200.github.io/xDrip/docs/xDrip-Version

**Reply by roco5** [reply]
Ok - but the phone I couldn't export from (despite a lot of attempts as the digitizer slowly failed from top to bottom), was the an Android phone running Android 12 and xdrip 051922  - Google Pixel 4a (impossible to take a screenshot from at this point as the screen is fried).    We just purchased a Google Pixel 6a and I have the Google Pixel 5a as a follower.

I assume you want to start here because those glucose alert settings come over with the Export DB? 
![image](https://user-images.githubusercontent.com/6946920/193870999-94d8754b-2b49-4d80-af1e-26e2303bb43b.png)

**Reply by Navid200** [reply]
Yes,
And there is no way you cannot export from a Pixel.

Where did you expect the database to be exported to?

All the details are here on how to export:
https://navid200.github.io/xDrip/docs/Backup-Database

This explains how to restore:
https://navid200.github.io/xDrip/docs/Restore-Database

However, restoring a database deletes the existing database before restoring.  So, if you have collected any new readings since you backed up the database, you will lose all of those readings.

**Reply by roco5** [reply]
So short answer is the export should work for bringing over these settings?

Q:  if we are transferring phones, better to use export import DB than the QR code for all settings?

**Reply by Navid200** [reply]
When you buy a new phone, it is really a mistake to transfer the settings.
The best course of action is to use the Source Wizard, which sets up the phone for you.
There are some settings that re different depending on the Android version.  if you are going from Android 9 to a new phone that is Android 11, transferring the settings would be the worst thing you can do.

The database contains the readings and the alerts.

The settings file contains everything else.
For example, if you hate the default colors and have customized all the colors, the only way to bring all those customizations is to transfer the settings file (QR code).  

I have a link for transferring to a new phone here:
https://navid200.github.io/xDrip/docs/New-Phone
When I buy a new phone, I only transfer the database.  
That brings in all my alerts.  However, I will still need to make sure my custom sounds are transferred.  Transferring the database does not bring the custom sounds files.  

The alerts are very important.  I actually like to go over them all on the new phone making sure that they work.  So, I'm not really in favor of pressing a button on the new phone and being done with it anyway.

**Comment by Navid200** [comment]
Now, with respect to transferring  alerts, I would like us to have the ability to only transfer the alerts without the readings.
I'm not sure how easy that is.

**Reply by roco5** [reply]
Yes - I don't need the readings often times, but would be great to have the alerts come over with the QR code.

**Comment by steve8x8** [comment]
Even transferring between identical phones (same model, same build of LineageOS) seems to get the sound assignments wrong, even when using the xml file which works more reliably than the QR code for me. (S4mini GT-i9195, LOS 18.1, MindTheGapps, if that matters.)

**Reply by Navid200** [reply]
Even if we add a feature that would allow user to only export/import alerts, and if I used it, I would still go over every single alert to verify that it behaves as I expect.
I cannot speak for all developers.  But, this may actually be intentional so that people manually create their alerts.

It's not getting a color wrong on the screen and not clearly seeing a color.  It's about not getting your low blood glucose alert messed up, so that you are warned that your blood glucose is very low.  That is very critical and the users should take it very seriously.

I have to confess the more I think about this, the less I like it.

**Reply by steve8x8** [reply]
You're making a good point here - but current behaviour is actually worse: it will assign *random* sounds to the alerts, *without* the user notifying (and somehow one would expect the new system, with the copied settings, to "just work"). So either get the sounds right, or tell the user after importing settings (QR and XML) to actually look into each and every alert.
Having all sounds listed somewhere (an extra system status page perhaps?) might help.

**Reply by Navid200** [reply]
I hear you.

Sometimes, we put requests on hold because there are serious shortcomings that get priority.
If xDrip alert did not alert you after transferring to a new phone, I would consider it a disaster and something that needed to be addressed right away.
But, the alert woks fine.  It just uses the standard xDrip sound.  So, the first time you hear it, you will know that you may want to change the sound as you did on your previous phone.
I consider this a good to have.

As an example of what I consider a disaster, if you lower your phone volume, your missed readings alert will not make a sound!  That's a serious problem and we have an open issue for it:
https://github.com/NightscoutFoundation/xDrip/issues/2122
